warn/warn.go

```diff
@@ -344,7 +344,31 @@ func outOfOrderLoadWarning(f *build.File, fix bool) []*Finding {
 		isExplicitRepo2 := strings.HasPrefix(load2Label, "@")
 		if isExplicitRepo1 == isExplicitRepo2 {
 			// Either both labels have explicit repository names or both don't, compare lexicographically
-			return load1Label < load2Label
+			canonicalModule1 := strings.TrimLeft(load1Label, "@")
+			canonicalModule2 := strings.TrimLeft(load2Label, "@")
+
+			module1Parts := strings.Split(canonicalModule1, ":")
+			package1, filename1 := module1Parts[0], module1Parts[1]
+			module2Parts := strings.Split(canonicalModule2, ":")
+			package2, filename2 := module2Parts[0], module2Parts[1]
+
+			// in case both loads lack packages, compare by file name
```

Comment:
Do you need these two checks or can you go straight to the package1 != package2 case? (delete 355 - 358)

If both packages are identical (including both empty), it would go to the `return filename1 < filename2`; but if only one package is empty, wouldn't the `return package1 < package2` still do the right thing?